tests/core/vm/test_computation.py

```diff
@@ -0,0 +1,184 @@
+        computation.extend_memory(1, -1)
+    # Stays the same if size is 0
+    assert len(computation.memory.bytes) == 0
+    computation.extend_memory(1, 0)
+    assert len(computation.memory.bytes) == 0
+    # Memory extends by 32 bytes and costs 3 gas
+    assert computation.gas_meter.gas_remaining == 100
+    computation.extend_memory(0, 1)
+    assert len(computation.memory.bytes) == 32
+    assert computation.gas_meter.gas_remaining == 97
+    # Memory isn't addded until 32 bytes are used
+    computation.extend_memory(1, 1)
+    assert len(computation.memory.bytes) == 32
+    computation.extend_memory(2, 32)
+    assert len(computation.memory.bytes) == 64
+    assert computation.gas_meter.gas_remaining == 94
+
+
+def test_register_accounts_for_deletion(computation):
+    # Fails if address isn't canonical
+    with pytest.raises(ValidationError):
+        computation.register_account_for_deletion("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6")
+    # Adds given address to accounts_to_delete
+    computation.register_account_for_deletion(CANONICAL_ADDRESS_1)
+    assert computation.accounts_to_delete[computation.msg.storage_address] == CANONICAL_ADDRESS_1
+    # An account cannot be registered for deletion multiple times
+    with pytest.raises(ValueError):
+        computation.register_account_for_deletion(CANONICAL_ADDRESS_1)
+    # Another account can be registered for deletion
+    computation.msg.storage_address = CANONICAL_ADDRESS_2
+    computation.register_account_for_deletion(CANONICAL_ADDRESS_1)
+
+
+def test_get_accounts_for_deletion(computation):
+    # Accounts for deletion starts empty
+    assert computation.get_accounts_for_deletion() == ()
+    computation.register_account_for_deletion(CANONICAL_ADDRESS_1)
+    # Get accounts for deletion returns the correct account
+    assert computation.get_accounts_for_deletion() == ((CANONICAL_ADDRESS_1, CANONICAL_ADDRESS_1),)
+    # Get accounts for deletion can return multiple accounts
+    computation.msg.storage_address = CANONICAL_ADDRESS_2
+    computation.register_account_for_deletion(CANONICAL_ADDRESS_2)
+    accounts_for_deletion = sorted(computation.get_accounts_for_deletion(),
+                                   key=lambda item: item[0])
+    assert CANONICAL_ADDRESS_1 == accounts_for_deletion[0][0]
+    assert CANONICAL_ADDRESS_2 == accounts_for_deletion[1][0]
+
+
+def test_add_log_entry(computation):
+    # Log entries starts empty
+    assert computation.log_entries == []
+    with pytest.raises(ValidationError):
+        # Fails if address isn't canonical
+        computation.add_log_entry("0x0f572e5295c57f15886f9b263e2f6d2d6c7b5ec6", [1, 2, 3], b'1')
+        # Fails if topic elements aren't uint256
+        computation.add_log_entry(CANONICAL_ADDRESS_1, [-1, 2, 3], b'1')
+        computation.add_log_entry(CANONICAL_ADDRESS_1, ['1', 2, 3], b'1')
+        # Fails if data isn't bytes
+        computation.add_log_entry(CANONICAL_ADDRESS_1, [1, 2, 3], 1)
```

Comment:
It would be good if this tried to add a non-bytes type string as well to see that it rejects string types that are not `bytes`.